### ISSUE TITLE
CW Attack not working as desired

### ISSUE THREAD

**Issue by Beemd** [Harry24k/adversarial-attacks-pytorch#32]
I am using CW targeted attack on a Convolution Neural Network trained for FashionMNIST. However I do not see any change in accuracy after the CW attack. I have tried using c =1, 10 and 100. I also tried using different learning rates like 0.001 and 0.0001. I have used different steps, like 100 and 1000.
However when I use targeted FGSM and PGD attacks, those work fine.

I need your support to help me figure out why the CW attack isn't working.
File with my code is attached.
[FMNIST_CNN.txt](https://github.com/Harry24k/adversarial-attacks-pytorch/files/6713879/FMNIST_CNN.txt)

**Comment by Harry24k** [comment]
Sorry for the late reply.
I checked `CW.py` and found the error when the targeted label is used.
I fixed as in https://github.com/Harry24k/adversarial-attacks-pytorch/commit/a3193194746bb348cc13649a9d931e58bc1e3751.
Sorry for the confusion and thank you for using the repo.
